hamilflow/models/harmonic_oscillator.py

```diff
@@ -63,8 +64,112 @@ class HarmonicOscillatorIC(BaseModel):
+    """
+
+    def __init__(
+        self,
+        system: Dict[str, float],
+        initial_condition: Optional[Dict[str, float]] = {},
+    ):
+        self.system = HarmonicOscillatorSystem.model_validate(system)
+        self.initial_condition = HarmonicOscillatorIC.model_validate(initial_condition)
+
+    @cached_property
+    def definition(self) -> Dict[str, float]:
+        """model params and initial conditions defined as a dictionary."""
+        return {
+            "system": self.system.model_dump(),
+            "initial_condition": self.initial_condition.model_dump(),
+        }
+
+    @abstractmethod
+    def _x(self, t: Union[float, np.ndarray]) -> Union[float, np.ndarray]:
+        r"""Solution to simple harmonic oscillators."""
+        ...
+
+    def __call__(self, n_periods: int, n_samples_per_period: int) -> pd.DataFrame:
+        """Generate time series data for the harmonic oscillator.
+
+        Returns a list of floats representing the displacement at each time step.
+
+        :param n_periods: Number of periods to generate.
+        :param n_samples_per_period: Number of samples per period.
+        """
+        time_delta = self.system.period / n_samples_per_period
+        time_steps = np.arange(0, n_periods * n_samples_per_period) * time_delta
+
+        data = self._x(time_steps)
+
+        return pd.DataFrame({"t": time_steps, "x": data})
+
+
+class SimpleHarmonicOscillator(HarmonicOscillatorBase):
+    r"""Generate time series data for a
+    [simple harmonic oscillator](https://en.wikipedia.org/wiki/Harmonic_oscillator).
+
+
+    In a one dimensional world, a mass $m$, driven by a force $F=-kx$, is described as
+
+    $$
+    \begin{align}
+    F &= - k x \\
+    F &= m a
+    \end{align}
+    $$
+
+    The mass behaves like a simple harmonic oscillator.
+
+    In general, the solution to a simple harmonic oscillator is
+
+    $$
+    x(t) = A \cos(\omega t + \phi),
+    $$
+
+    where $\omega$ is the angular frequency, $\phi$ is the initial phase, and $A$ is the amplitude.
+
+
+    To use this generator,
+
+    ```python
+    params = {"omega": omega}
+
+    ho = SimpleHarmonicOscillator(params=params)
+
+    df = ho(n_periods=1, n_samples_per_period=10)
+    ```
+
+    `df` will be a pandas dataframe with two columns: `t` and `x`.
+    """
+
+    def __init__(
+        self,
+        system: Dict[str, float],
+        initial_condition: Optional[Dict[str, float]] = {},
+    ):
+        super().__init__(system, initial_condition)
+        if self.system.type != "simple":
+            raise ValueError(
+                f"System is not a Simple Harmonic Oscillator: {self.system}"
+            )
+
+    def _x(self, t: Union[float, np.ndarray]) -> Union[float, np.ndarray]:
+        r"""Solution to simple harmonic oscillators:
+
+        $$
+        x(t) = x_0 \cos(\omega t).
+        $$
+        """
+        return self.initial_condition.x0 * np.cos(self.system.omega * t)
```

Comment:
Initial phase is missing

Reply:
https://github.com/kausalflow/hamilflow/pull/48/commits/0352c4e5253ef713029b4a5727aa72b679c088c0

Reply:
I am a bit hesitate to add the initial phase in general. It introduces more terms in the damped case.
Here I only adjusted it for the simple harmonic oscillators. 

Maybe something to think about:
Either we adjust the damped oscillator case to include the initial phase or we just say that, for our use case, we don't really care about initial phase since it is a time translation for simple harmonic oscillators.